src/Symfony/Component/Process/Pipes/WindowsPipes.php

```diff
@@ -52,27 +52,30 @@ public function __construct($disableOutput, $input)
                 Process::STDERR => Process::ERR,
             );
             $tmpDir = sys_get_temp_dir();
-            if (!@fopen($file = $tmpDir.'\\sf_proc_00.check', 'wb')) {
-                throw new RuntimeException('A temporary file could not be opened to write the process output to, verify that your TEMP environment variable is writable');
-            }
-            @unlink($file);
+            $error = '';
+            set_error_handler(function ($type, $msg) use (&$error) { $error = $msg; });
             for ($i = 0;; ++$i) {
                 foreach ($pipes as $pipe => $name) {
                     $file = sprintf('%s\\sf_proc_%02X.%s', $tmpDir, $i, $name);
-                    if (file_exists($file) && !@unlink($file)) {
+                    if (file_exists($file) && !unlink($file)) {
                         continue 2;
                     }
-                    $h = @fopen($file, 'xb');
+                    $h = fopen($file, 'xb');
+                    if (!$h && false === strpos($error, 'File exists')) {
+                        restore_error_handler();
+                        throw new RuntimeException('A temporary file could not be opened to write the process output to, verify that your TEMP environment variable is writable');
```

Comment:
If wé can, then I'd even prefer: `sprintf(''A temporary file could not be opened to write the process output: %s', $error)`